src/js/containers/award/shared/ContractGrantActivityContainer.jsx

```diff
@@ -42,42 +43,61 @@ const ContractGrantActivityContainer = ({ awardId, awardType, dates }) => {
+                const currentTransactionIndex = acc.findIndex((x) => x.action_date.valueOf() === updatedData.action_date.valueOf());
+                /**
+                 * When we have multiple transactions on the same day, we will sum their obligation and
+                 * we will keep track of all the transactions on the same date for the tooltips in the
+                 * allTransactions property.
+                 */
+                if (currentTransactionIndex !== -1) {
+                    // update the allTransactions array if it exists
+                    if (acc[currentTransactionIndex]?.allTransactions) {
+                        acc[currentTransactionIndex].allTransactions.push(updatedData);
+                    }
+                    else {
+                        /**
+                         * The first duplicate found.
+                         * We will add the duplicate which is this data, and we will add the
+                         * original node which is acc[currentTransactionIndex].
+                         */
+                        const clonedTransaction = cloneDeep(acc[currentTransactionIndex]);
+                        acc[currentTransactionIndex].allTransactions = [clonedTransaction, updatedData];
+                    }
                     /**
-                     * The first duplicate found.
-                     * We will add the duplicate which is this data, and we will add the
-                     * original node which is acc[currentTransactionIndex].
+                     * We sum the obligation last since we will want to keep the original obligation
+                     * value if we add it to the allTransactions array.
                      */
-                    const clonedTransaction = cloneDeep(acc[currentTransactionIndex]);
-                    acc[currentTransactionIndex].allTransactions = [clonedTransaction, updatedData];
+                    const sumOfObligations = acc[currentTransactionIndex].federal_action_obligation + updatedData.federal_action_obligation;
+                    acc[currentTransactionIndex].federal_action_obligation = sumOfObligations;
+                    return acc;
                 }
-                /**
-                 * We sum the obligation last since we will want to keep the original obligation
-                 * value if we add it to the allTransactions array.
-                 */
-                const sumOfObligations = acc[currentTransactionIndex].federal_action_obligation + updatedData.federal_action_obligation;
-                acc[currentTransactionIndex].federal_action_obligation = sumOfObligations;
+                acc.push(updatedData);
                 return acc;
-            }
-            acc.push(updatedData);
-            return acc;
-        }, []);
-        // remove negative values
-        newData = newData.filter((data) => !data.federal_action_obligation.toString().startsWith('-'));
```

Comment:
Looks like we need to keep this; below on line `85` we're only removing transactions w/o dates.

Reply:
We want to keep any transaction that has a negative summed federal_action_obligation.  The `isBadData` method uses these to determine if we should draw the chart.

```
    // Bad Data - hook - run on mount and if award changes
    useEffect(() => {
        setBadDates(isBadData(dates, awardType, transactions));
    }, [
        dates,
        awardType,
        transactions
    ]);
```